### PR TITLE
Set pm2 log paths based on network - Closes #168

### LINKED ISSUE #168: Set pm2 log paths based on $NETWORK
Edit the pm2-lisk.json file when starting for creating the logs directories and the pm2 config file so pm2 logs app and err to the same directory as the application 

Related to: https://github.com/LiskHQ/lisk/issues/2452

### PR DESCRIPTION
**Pull request by pablitovicente**
@Nazgolze @fchavant I added a Node script that will create directories, based on the NETWORK name, and edit the pm2-lisk.json file to point to the correct logs directory.

### CODE REVIEW

**Comment by fchavant** [comment]
@pablitovicente can't we use a js configuration file? (https://pm2.io/doc/en/runtime/guide/ecosystem-file/)

**Comment by fchavant** [comment]
@pablitovicente my impression was that you use `os.environ` in the JS file.

**Comment by pablitovicente** [comment]
> @pablitovicente my impression was that you use `os.environ` in the JS file.

Oh will check if possible to use `process.env.NETWORK || <some default>`

**Comment by pablitovicente** [comment]
@fchavant changing to a js config works as I can access process.env.<ENV VARIABLE HERE> but the catch is that many of the scripts relay on JQ and the existing JSON configuration file so changing to a JS file will require making several changes that I think exceed the scope of this issue. We might want to discuss more alternatives?

**Comment by pablitovicente** [comment]
@fchavant @Nazgolze done with jq now. Much more cleaner I think! Please review and let me know of any feedback.

File: packaged/lisk.sh [hunk 1]
```diff
@@ -285,6 +286,12 @@ start_lisk() {
 	fi
 }
 
+setLogspath() {
+	jq '.apps[0].out_file = "logs/\(env.LISK_NETWORK)/lisk.app.log"' < etc/pm2-lisk.json|jq '.apps[0].error_file = "logs/\(env.LISK_NETWORK)/lisk.app.err"' > tmp.json
```
Comment (fchavant) [inline packaged/lisk.sh]: - use `$LISK_NETWORK` variable
- invoke `jq` only once
- use `mktemp`

```suggestion
	TEMP=$( mktemp )
	jq '.apps[0].out_file = "logs/'"$LISK_NETWORK"'/lisk.app.log"|.apps[0].error_file = "logs/'"$LISK_NETWORK"'/lisk.app.err"' "$PM2_CONFIG" >"$TEMP"
	mv -f "$TEMP" "$PM2_CONFIG"
```

File: packaged/lisk.sh [hunk 2]
```diff
@@ -276,6 +276,7 @@ stop_redis_cmd(){
 
 start_lisk() {
 	start_redis
+	setLogspath
```
Comment (fchavant) [inline packaged/lisk.sh]: Not sure this warrants a function (not used anywhere else)

File: packaged/lisk.sh [hunk 3]
```diff
@@ -285,6 +286,12 @@ start_lisk() {
 	fi
 }
 
+setLogspath() {
```
Comment (fchavant) [inline packaged/lisk.sh]: stick to snake_case

Comment (pablitovicente) [reply packaged/lisk.sh]: good catch!

File: packaged/lisk.sh (hunk 2 above)
Comment (pablitovicente) [reply packaged/lisk.sh]: Sure is not reused but at least it's isolated on it's own

**Comment by pablitovicente** [comment]
@fchavant changes made comment about function answered let me know what you think.

**Comment by fchavant** [comment]
@pablitovicente looks good; I think we need to ensure the `"$( pwd )/logs/$LISK_NETWORK"` directory exists though.

File: packaged/lisk.sh [hunk 4]
```diff
@@ -285,6 +286,15 @@ start_lisk() {
 	fi
 }
 
+set_logs_path() {
+	if [ ! -d "$(pwd)/logs/$LISK_NETWORK" ]; then
```
[the remark this answers is not in the document]
Comment (pablitovicente) [reply packaged/lisk.sh]: @fchavant check added please check.

**Pull request merged**